submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/service/ParentInfoService.kt

```diff
@@ -2,26 +2,49 @@ package ac.uk.ebi.biostd.submission.service
 
 import ac.uk.ebi.biostd.persistence.common.service.SubmissionMetaQueryService
 import ac.uk.ebi.biostd.submission.exceptions.CollectionInvalidAccessTagException
-import ebi.ac.uk.model.constants.SubFields
+import ac.uk.ebi.biostd.submission.validator.collection.CollectionValidator
+import ebi.ac.uk.base.ifFalse
+import ebi.ac.uk.extended.model.ExtSubmission
+import ebi.ac.uk.extended.model.isCollection
+import ebi.ac.uk.model.constants.SubFields.PUBLIC_ACCESS_TAG
+import org.springframework.beans.factory.BeanFactory
 import java.time.OffsetDateTime
 
-class ParentInfoService(private val queryService: SubmissionMetaQueryService) {
+class ParentInfoService(
+    private val beanFactory: BeanFactory,
+    private val queryService: SubmissionMetaQueryService
+) {
     fun getParentInfo(parentAccNo: String?): ParentInfo = when (parentAccNo) {
         null -> ParentInfo(emptyList(), null, null)
         else -> parentInfo(parentAccNo)
     }
 
     private fun parentInfo(parentAccNo: String): ParentInfo {
-        val project = queryService.getBasicCollection(parentAccNo)
-        return ParentInfo(accessTags(parentAccNo), project.releaseTime, project.accNoPattern)
+        val parentInfo = collectionInfo(parentAccNo)
+
+        return ParentInfo(accessTags(parentAccNo), parentInfo.releaseTime, parentInfo.accNoPattern)
     }
 
     private fun accessTags(parentAccNo: String): List<String> {
-        val accessTags = queryService.getAccessTags(parentAccNo).filterNot { it == SubFields.PUBLIC_ACCESS_TAG.value }
+        val accessTags = queryService.getAccessTags(parentAccNo).filterNot { it == PUBLIC_ACCESS_TAG.value }
         require(accessTags.contains(parentAccNo)) { throw CollectionInvalidAccessTagException(parentAccNo) }
 
         return accessTags
     }
+
+    fun executeCollectionValidators(submission: ExtSubmission) = submission.isCollection.ifFalse {
+        submission
+            .collections
+            .mapNotNull { collectionInfo(it.accNo).validator }
+            .forEach { validate(it, submission) }
+    }
+
+    private fun collectionInfo(accNo: String) = queryService.getBasicCollection(accNo)
+
+    private fun validate(validator: String, submission: ExtSubmission) =
```

Comment:
this method is not doing what is said that is does making logic harder to follow, it does not validate but it gets the validation bean and validate (it is also doing to two operations and not a single one)